backend/src/main/java/xbdd/util/StatusHelper.java

```diff
@@ -42,15 +42,15 @@ public static Statuses reduceStatuses(final List<String> allStatuses) {
 
 	// go through all the steps in a scenario and reduce to a status for the scenario.
 	public static Statuses getFinalScenarioStatus(final DBObject scenario, final boolean includeManualResults) {
-		final List<String> allStatuses = new ArrayList<String>();
+		final List<String> allStatuses = new ArrayList<>();
 		final BasicDBList steps = (BasicDBList) scenario.get("steps");
 		if (includeManualResults) { // if we have got a bunch of manual step executions
 			boolean hasManuallyExecutedSteps = false;
-			final List<String> manualSteps = new ArrayList<String>();
+			final List<String> manualSteps = new ArrayList<>();
 			// go through each step creating an array as though they were manual
 			if (steps != null) {
-				for (int i = 0; i < steps.size(); i++) {
-					final DBObject step = (DBObject) steps.get(i);
+				for (Object o : steps) {
+					final DBObject step = (DBObject) o;
```

Comment:
Far out this code is in desperate need of DTOs!

Reply:
Yup, I removed all the warning suppressions to highlight those cases more.